google-cloud-core/src/main/java/com/google/cloud/Timestamp.java

```diff
@@ -74,6 +79,37 @@ public static Timestamp ofTimeSecondsAndNanos(long seconds, int nanos) {
   }
 
   /**
+   * Creates an instance representing the value of {@code microseconds}
+   *
+   * @throws IllegalArgumentException if the timestamp is outside the representable range
+   */
+  public static Timestamp ofTimeMicroseconds(long microseconds) {
+    long seconds = microseconds / 1000000L;
```

Comment:
Use `TimeUnit.MICROSECONDS.toSeconds(microseconds)`, and a similar type of conversion on the next line, and in the `of(Date)` function.